src/erlzk_conn.erl

```
@@ -539,14 +551,20 @@ maybe_store_watcher(Code, {Op, _From, Path, Watcher}, Watchers) ->
         true -> store_watcher(Op, Path, Watcher, Watchers)
     end.
 
-notify_callers_closed(State=#state{reqs=Reqs}) ->
-    notify_callers_closed(dict:to_list(Reqs)),
-    State#state{reqs=dict:new()};
-notify_callers_closed([]) ->
+notify_callers_closed(State=#state{reqs=Reqs, auths=Auths}) ->
+    lists:foreach(fun notify_req_closed/1, dict:to_list(Reqs)),
+    lists:foreach(fun notify_auth_closed/1, queue:to_list(Auths)),
```

so before this, add_auth can only get timeouts in case of state reset e.g. reconnect ?
good catch

---

Yes, and auths were also silently dropped in case of a reconnect.